### PR TITLE
Cleanup ms unmerged t2 ch cern

### LINKED ISSUE #10701: MSUnmerged: Add sub tree filters
**Impact of the new feature**
MSUnmerged

**Is your feature request related to a problem? Please describe.**
In the process  of cleaning T2_CH_CERN we found that it is worth having the ability to exploit some filters on the sub tree inside the `/store/unmerged/` area.

**Describe the solution you'd like**
To provide a method to include and exclude parts of the tree inside the `/store/unmerged` area such as:
```
/store/unmerged/tier0
/store/unmerged/data/Tier0
/store/unmerged/logs/prod/
``` 
etc. 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by todor-ivanov**
Fixes #10701 

#### Status
ready

#### Description
Providing logic to Include or Exclude some part of the tree bellow the `/store/unmerged/` area

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
None

#### External dependencies / deployment changes
None

### CODE REVIEW

File: src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py [hunk 1]
```diff
@@ -488,9 +490,30 @@ def genFunc(pattern, iterable):
...
 
+        # NOTE: If the 'dirFilterIncl' is non empty then the cleaning process will
+        #       be enclosed only in this part of the tree and will ignore anything
+        #       from /store/unmerged/ which does not belong to the included filter
+        # NOTE: 'dirFilterExcl' is always applied.
+
+        # Merge the additional filters into a final set to be applied:
+        dirFilterIncl = set(self.msConfig['dirFilterIncl'])
+        dirFilterExcl = set(self.msConfig['dirFilterExcl'])
+        dirFilterAll = dirFilterIncl - dirFilterExcl
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py:501]: This won't work as expected:
```
In [8]: set([]) - set(["malta"])
Out[8]: set()
```

Maybe the best would be to:
* if an include dir is provided: simply match that dir against the RSE dump, forget all the rest (including any possible directory exclusion). Honestly speaking, we will likely NEVER use this option once the service is stable in production.
* else-if an exclude dir is provided, match everything from the RSE dump but the exclude directories.

Note the `else-if` though.

Comment (amaltaro) [reply src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py:501]: Todor, as chatted over slack, this code already takes the exclusion list into account (and higher precedence over the inclusion list). Thanks!

**Comment by todor-ivanov** [comment]
thanks @amaltaro  I've just squashed those tree commits.

**Comment by todor-ivanov** [comment]
@amaltaro for some reaon I cannot request your review again, but please take a look at the latest commit which provides a fix for the eventual bug of not properly applying the exclusionlist

**Comment by amaltaro** [comment]
Unit tests helped me spotting a flaw in the including + excluding logic. Example setup is:
Incl: '/store/unmerged/data/prod/2018/'
Excl: '/store/unmerged/data/prod'

with these 2 LFNs:
```
                "/store/unmerged/data/prod/2018/1/12/log6.tar",
                "/store/unmerged/data/prod/2019/1/12/log7.tar",
```

it would let the 2018 file to go through. I'm getting this fixed in a separate branch, but code will be more complex to ensure excluding list is always properly evaluated and has precedence over the inclusion list.

**Comment by amaltaro** [comment]
Superseeded by https://github.com/dmwm/WMCore/pull/10706
Closing this one.

**Pull request closed without merging**